agent/ecscni/plugin.go

```diff
@@ -145,37 +214,70 @@ func (client *cniClient) constructNetworkConfig(cfg *Config) (*libcni.NetworkCon
 		BlockInstanceMetdata: cfg.BlockInstanceMetdata,
 	}
 
-	bridgeConfBytes, err := json.Marshal(bridgeConf)
+	eniConfBytes, err := json.Marshal(eniConf)
 	if err != nil {
-		seelog.Errorf("Marshal bridge configuration failed, error: %v", err)
+		seelog.Errorf("Marshal eni configuration error: %v", err)
 		return nil, err
 	}
-	plugins := []*libcni.NetworkConfig{
-		&libcni.NetworkConfig{
-			Network: &types.NetConf{
-				Type: ECSBridgePluginName,
-			},
-			Bytes: bridgeConfBytes,
+
+	networkConfig := &libcni.NetworkConfig{
+		Network: &types.NetConf{
+			Type: ECSENIPluginName,
 		},
+		Bytes: eniConfBytes,
 	}
 
-	eniConfBytes, err := json.Marshal(eniConf)
+	return networkConfig, nil
+}
+
+func (client *cniClient) createIPAMConfig(cfg *Config) (*libcni.NetworkConfig, error) {
+	ipam := IPAMConfig{
+		Type:        ECSIPAMPluginName,
+		CNIVersion:  client.cniVersion,
+		IPV4Subnet:  client.subnet,
+		IPV4Address: cfg.IPAMV4Address,
+		ID:          cfg.ID,
+	}
+	ipamConfig := IPAMNetworkConfig{
+		Name:       ECSIPAMPluginName,
+		CNIVersion: client.cniVersion,
+		IPAM:       ipam,
+	}
+
+	ipamConfigBytes, err := json.Marshal(ipamConfig)
 	if err != nil {
-		seelog.Errorf("Marshal eni configuration error: %v", err)
-		return nil, err
+		return nil, errors.Wrapf(err, "Marshal ipam plugin configuration failed")
 	}
-	plugins = append(plugins, &libcni.NetworkConfig{
+
+	networkConfig := &libcni.NetworkConfig{
 		Network: &types.NetConf{
-			Type: ECSENIPluginName,
+			Type: ECSIPAMPluginName,
 		},
-		Bytes: eniConfBytes,
-	})
+		Bytes: ipamConfigBytes,
+	}
 
-	netconf := &libcni.NetworkConfigList{
+	return networkConfig, nil
+}
+
+// createAddNetworkConfig creates the configuration for invoking plugins
+func (client *cniClient) createNetworkConfig(cfg *Config, bridgeConfigFunc func(*Config) (*libcni.NetworkConfig, error)) (*libcni.NetworkConfigList, error) {
```

Comment:
It looks like this function is the entrypoint for both `createBridgeConfig` and `createBridgeConfigWithENI`.  Can you move it above those functions?